modules/hostap/src/supp_api.c

```diff
@@ -951,11 +1187,36 @@ int supplicant_ap_enable(const struct device *dev,
 
 int supplicant_ap_disable(const struct device *dev)
 {
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+	struct hostapd_iface *iface;
+	int ret = 0;
+#else
```

Comment:
Vendors can choose to setup AP with hostapd by enabling this flag. Or, the AP will be setup by wpa_supplicant. 
Also, I found that there are many files in Zephyr has `#ifdef` in .c files, such as net_if.c, tcp.c. So I think it might be ok if I put `#ifdef` here?

Reply:
Obviously, conditional compilation is needed. However, I am afraid this file become a nightmare to maintain with `#ifdef` inside.

I suggest to split this file in two (or three). Then, rely on CMakeLists.txt (using `zephyr_library_sources_ifdef()`) to manage conditional compilation.

Reply:
Sure, got it, may be `hapd_api.c` and then move all hostapd related code to that file.

Reply:
Hi @krish2718 
if split supp_api.c to hapd_api.c, do you think supp_main.c and supp_events.c needs same split?

the reason I said a lot work to do is on our internal code base, because  following feature which already finished, such as: uap enterpise,  ap dpp are all done based on supp_api.c (none split file),
so there will be a lot of rebase work to do in our internal code base

do you think it is possible that we first finish upstream all our PRs in this none split file, and finally move to a split file
for supp_api.c, supp_events.c and supp_main.c?

Reply:
Fine with me, we can add an improvement in GH to track.

Reply:
Hi @jerome-pouiller , do you agree with our above plan plan? could you pls take time to comment?
thanks in advance

Reply:
Hi @jerome-pouiller 
any feedback/comments on following plan ?
 if split supp_api.c to hapd_api.c, do you think supp_main.c and supp_events.c needs same split?

the reason I said a lot work to do is on our internal code base, because target to upstream features which already finished, such as: uap enterpise, ap dpp are all done based on supp_api.c (none split file), so there will be a lot of rebase work to do in our internal code base

do you think it is possible that we first finish upstream all our PRs in this none split file, and finally move to a split file for supp_api.c, supp_events.c and supp_main.c in one Pull Request ?